commands/delete/cluster/command.go

```diff
@@ -185,32 +186,63 @@ func printResult(cmd *cobra.Command, args []string) {
 	}
 }
 
+func getClusterID(clusterNameOrID string, clientWrapper *client.Wrapper) (string, error) {
+	auxParams := clientWrapper.DefaultAuxiliaryParams()
+	auxParams.ActivityName = listClustersActivityName
+
+	response, err := clientWrapper.GetClusters(auxParams)
+	if err != nil {
+		if clienterror.IsUnauthorizedError(err) {
+			return "", microerror.Mask(errors.NotAuthorizedError)
+		}
+		if clienterror.IsAccessForbiddenError(err) {
+			return "", microerror.Mask(errors.AccessForbiddenError)
+		}
+
+		return "", microerror.Mask(err)
+	}
+
+	for _, cluster := range response.Payload {
+		if cluster.ID == clusterNameOrID || cluster.Name == clusterNameOrID {
```

Comment:
Here we basically go for the first found match right?

I think since names do not have to be unique, we should fail with a "Cluster name is ambiguous" type of failure message when the name you chose is present more than once.

It would be pretty rare, but I can imagine someone naming their clusters "Dev" "Production" and "Staging" and then having multiple organisations with those clusters all with the same name or something like that. So it's not too far fetched.

Reply:
Ahhh yesss, I actually had this in my notes but completely forgot about this

Thank you for noticing, I'll update and throw an error if there are multiple clusters with the same names

Reply:
Yep, even if rare, it would be fatal, so I consider that absolutely necessary.